ui/src/timeMachine/reducers/index.ts

```diff
@@ -142,6 +135,40 @@ export const initialState = (): TimeMachinesState => ({
   },
 })
 
+const getTableProperties = (view, files) => {
```

Comment:
this is mostly a csv parser for now. since the csv can be very large, we dont want to make a lot of copies of it in memory, so we move a pointer around in place and try to extract only the line that has the meta data in it that we want